openmc/mgxs_library.py

```diff
@@ -1634,6 +1685,267 @@ def set_inverse_velocity_mgxs(self, inverse_velocity, temperature=294.,
+            domain is subdivided into. This is required when
+            :param:`target_representation` is "angle".
+
+        Returns
+        -------
+        openmc.XSdata
+            Multi-group cross section data with the same data as self, but
+            represented as specified in :param:`target_representation`.
+
+        """
+
+        check_value('target_representation', target_representation,
+                    _REPRESENTATIONS)
+        if target_representation == 'angle':
+            check_type('num_polar', num_polar, Integral)
+            check_type('num_azimuthal', num_azimuthal, Integral)
+            check_greater_than('num_polar', num_polar, 0)
+            check_greater_than('num_azimuthal', num_azimuthal, 0)
+
+        xsdata = copy.deepcopy(self)
+
+        # First handle the case where the current and requested
+        # representations are the same
+        if target_representation == self.representation:
+            # Check to make sure the num_polar and num_azimuthal values match
+            if target_representation == 'angle':
+                if num_polar != self.num_polar or num_azimuthal != self.num_azimuthal:
+                    raise NotImplementedError("XCannot translate between "
+                                              "`angle` representations with "
+                                              "different angle bin structures")
+            # Nothing to do as the same structure was requested
+            return xsdata
+
+        xsdata.representation = target_representation
+        # We have different actions depending on the representation conversion
+        if target_representation == 'isotropic':
+            # This is not needed for the correct functionality, but these
+            # values are changed back to None for clarity
+            xsdata._num_polar = None
+            xsdata._num_azimuthal = None
+        elif target_representation == 'angle':
+            xsdata.num_polar = num_polar
+            xsdata.num_azimuthal = num_azimuthal
+
+        # Reset xs_shapes so it is recalculated the next time it is needed
+        xsdata._xs_shapes = None
+
+        for i, temp in enumerate(xsdata.temperatures):
+            for xs in ['total', 'absorption', 'fission', 'nu_fission',
+                       'scatter_matrix', 'multiplicity_matrix',
+                       'prompt_nu_fission', 'delayed_nu_fission',
+                       'kappa_fission', 'chi', 'chi_prompt', 'chi_delayed',
+                       'beta', 'decay_rate', 'inverse_velocity']:
+                # Get the original data
+                orig_data = getattr(self, '_' + xs)[i]
+                if orig_data is not None:
+                    if target_representation == 'isotropic':
+                        # Since we are going from angle to isotropic, the
+                        # current data is just the average over the angle bins
+                        new_data = orig_data.mean(axis=(0, 1))
```

Comment:
I don't think this will conserve reaction rates since it isn't flux-averaged, correct? I'm not opposed to you using a simple arithmetic average here, but you should clearly state the caveat in the docstring that you are not using flux-weighting and hence reaction rates / reactivity will not be conserved.